Tasks/UseDotNetV2/versioninstaller.ts

```diff
@@ -38,10 +38,13 @@ export class VersionInstaller {
                 var downloadPath = await toolLib.downloadToolWithRetries(downloadUrl)
             }
             catch (ex) {
-                tl.warning(tl.loc("CouldNotDownload", downloadUrl, ex));
-                let fallBackUrl = `https://dotnetcli.azureedge.net/dotnet/${this.packageType === "runtime" ? "Runtime" : "Sdk"}/${version}/${downloadUrl.substring(downloadUrl.lastIndexOf('/') + 1)}`;
-                console.log("Using fallback url for download: " + fallBackUrl);
-                var downloadPath = await toolLib.downloadToolWithRetries(fallBackUrl)
+                try {
+                    tl.warning(tl.loc("CouldNotDownload", downloadUrl, ex));
+                    var downloadPath = await this.downloadFromFallbackUrl("https://builds.dotnet.microsoft.com/dotnet", this.packageType, version, downloadUrl);
+                } catch(ex) {
+                    tl.warning(tl.loc("CouldNotDownload", downloadUrl, ex));
```

Comment:
this line will report the wrong url for 'could not download', which should be builds.dotnet.microsoft.com

Reply:
updated